iib/workers/tasks/utils.py

```diff
@@ -175,28 +176,49 @@ def _requires_max_ocp_version(bundle):
     return False
 
 
-def serve_index_registry(db_path):
+def serve_registry(base_dir, from_index=None, fbc_dir=None, db_path=None):
     """
     Locally start OPM registry service, which can be communicated with using gRPC queries.
 
     Due to IIB's paralellism, the service can run multiple times, which could lead to port
     binding conflicts. Resolution of port conflicts is handled in this function as well.
 
+    :param str base_dir: base directory to create temporary files in.
+    :param str from_index: index image to inspect.
+    :param str fbc_dir: path to file-based catalog directory.
     :param str db_path: path to index database containing the registry data.
     :return: tuple containing port number of the running service and the running Popen object.
     :rtype: (int, Popen)
     :raises IIBError: if all tried ports are in use, or the command failed for another reason.
     """
+    from iib.workers.tasks.build import _get_index_database
+
     conf = get_worker_config()
     port_start = conf['iib_grpc_start_port']
     port_end = port_start + conf['iib_grpc_max_port_tries']
 
+    assert (
+        len([i for i in (from_index, fbc_dir, db_path) if i]) == 1
+    ), 'Only one keyword parameter can be set'
+
+    if from_index:
+        if not is_image_fbc(from_index):
+            db_path = _get_index_database(from_index, base_dir)
+        else:
+            fbc_dir = get_index_fbc_dir(from_index, base_dir)
```

Comment:
In case of merging the `get_index_fbc_dir` with `get_hidden_index_database` you should change its name here.

Also I still don't understand the use of `get_hidden_index_database` since I thought it would be used here but instead you're using `_get_index_database`.

Both `iib.workers.tasks.fbc_utils.get_hidden_index_database` and `iib.workers.tasks.build._get_index_database` seems redundant to me. I know they're getting the SQLite from different locations, the first one from hidden path set in configuration and the last one from the image's label, but do we really need that?

It seems we're replicating the very same functions over different places.

Reply:
@JAVGan `get_hidden_index_database` will be used only for File-Based Catalog images and since we have `if` on line above saying that the image is not FBC, then we should get the index.db from usual location.

Both are getting same SQLite database. Only for FBC image it should be hidden and used only for Phase 1 of this "project" we have to use this hidden database internally in IIB. We are expecting to remove behavior this in future.